realtek: build sane factory images for all DGS-1210 models

plappermaul opened the pull request:
Currently we build factory images only for DGS-1210-28 model. Relax
that constraint and take care about all models. The initial commit forgot
to label the image in the right way. Therefore WebUI and CLI show
strange version information. Fix this too.

```
Before:
DGS-1210-20> show firmware information
IMAGE ONE:
Version      : xfo/QE~WQD"A\Scxq...
Size         : 5505185 Bytes

After:
DGS-1210-20> show firmware information
IMAGE ONE:
Version      : OpenWrt
Size         : 5505200 Bytes
```

svanheule commented on target/linux/realtek/image/Makefile:
```diff
@@ -21,6 +21,11 @@ define Build/dlink-cameo
 
+define Build/dlink-version
+	echo -n "OpenWrt" >> $@
+	dd if=/dev/zero bs=1 count=25 >> $@
```
I usually do this the other way around `bs=25 count=1` for higher throughput. But I guess we're talking about minor performance gains again here :)

svanheule reviewed (commented):
I know the changes are rather small, but they are two different things. Can you split this into:
1. Build image for all devices, like currently done for the DGS-1210-28
2. Append a version trailer to the images

It makes reviewing easier, since I don't have to decipher which changes belong to which part of the description. Also, if bisecting is ever required, an issue can more easily be pinpointed to the first or second commit ("it was always broken" vs. "the version string breaks the image").

For the build-for-all change, you could also add the argument that this change is possible in the proposed way because the partition layouts are identical (shared DTSI).

plappermaul commented:
dont' break this one

svanheule commented on target/linux/realtek/image/rtl838x.mk:
```diff
@@ -23,7 +23,8 @@ define Device/d-link_dgs-1210
   IMAGE/factory_image1.bin := append-kernel | pad-to 64k | \
-        append-rootfs | pad-rootfs | pad-to 16 | check-size | dlink-headers
+	append-rootfs | pad-rootfs | pad-to 16 | check-size | \
+	dlink-version | dlink-headers
```
Looking at a FW image for the [DGS-1210-28](https://support.dlink.com/resource/PRODUCTS/DGS-1210-SERIES/REVF/FIRMWARE/v6.30.016/DGS-1210-28_REVF_FIRMWARE_v6.30.016.zip), the firmware version string + CAMEOTAG are 32 bytes. So I assume that's why you're adding "OpenWrt" + 25 bytes here.

Since the version string is written to flash, it should also be covered by `check-size`. `dlink-version` will only add 32 bytes, at a point where you're only 16 bytes into the next erase block, so the result should never be different though.

The `dlink-version` tag is appended to the data after the DEADCODE marker. So it doesn't look like this will survive the jffs2 initialisation for OpenWrt's rootfs_data partition. Is this something you wish to support?

svanheule commented:
The first commit has some stray German: "because auf common flash layout".

plappermaul committed:
realtek: build factory images for all DGS-1210 models

Currently we build factory images only for DGS-1210-28 model. Relax
that constraint and take care about all models. Tested on DGS-1210-20
and should work on other models too because of common flash layout.

Signed-off-by: Markus Stockhausen <markus.stockhausen@gmx.de>

plappermaul committed:
realtek: build sane factory images for DGS-1210 models

During upload of firmware images the WebUI and CLI patch process
extracts a version information from the uploaded file and stores it
onto the jffs2 partition. To be precise it is written into the
flash.txt or flash2.txt files depending on the selected target image.
This data is not used anywhere else. The current OpenWrt factory
image misses this label. Therefore version information shows only
garbage. Fix this.

Before:
DGS-1210-20> show firmware information
IMAGE ONE:
Version      : xfo/QE~WQD"A\Scxq...
Size         : 5505185 Bytes

After:
DGS-1210-20> show firmware information
IMAGE ONE:
Version      : OpenWrt
Size         : 5505200 Bytes

Signed-off-by: Markus Stockhausen <markus.stockhausen@gmx.de>

luizluca commented:
I would be nice to backport all recent DGS-1210 fixes to 22.03

svanheule commented:
Thanks, @luizluca! Can you provide a Tested-by:?

svanheule commented:
Feel free to open a PR with all the required parts. Patches should be picked with `git cherry-pick -x`.

luizluca commented:
> Thanks, @luizluca! Can you provide a Tested-by:?

Does a github comment count?

Tested-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>

luizluca commented:
> Regarding 22.03, feel free to open a PR with all the required parts. Patches should be picked with `git cherry-pick -x`.

I'll do once this PR gets merged. It seems to be the last piece to finally have a pretty good firmware format including factory images. I don't believe we'll have to change it again unless dlink requires encrypted images (.con format).

svanheule commented:
Thanks! Merged with:
2b49ec3a28ad realtek: build factory images for all DGS-1210 models
fae3ac356045 realtek: build sane factory images for DGS-1210 models
